### DISCUSSION TITLE
OpenSSL store API - attempt to implement delete() semantics

### DISCUSSION THREAD

**Discussion by beldmit** [openssl/openssl#21822] [General Discussion]
I'm trying to understand the store API to implement the delete functionality (e.g. for PKCS#11 provider) or at least to design it.

It's easy to add a new optional callback on the provider side (and no need to do it for legacy purposes). 

Do I correctly understand that OSSL_STORE_INFO is a proper argument to a hypothetical function `OSSL_STORE_delete`? I also feel some problems related to OSSL_STORE_CTX having 

```
    /*
     * Cache of stuff, to be able to return the contents of a PKCS#12
     * blob, one object at a time.
     */
    STACK_OF(OSSL_STORE_INFO) *cached_info;
```
because we may want to invalidate this cache.

**Comment by levitte** [comment]
Considering the delete functionality has now been merged, this discussion can be closed
